composer/utils/dist.py

```diff
@@ -420,3 +421,35 @@ def get_sampler(dataset: torch.utils.data.Dataset, *, drop_last: bool, shuffle:
         num_replicas=get_world_size(),
         rank=get_global_rank(),
     )
+
+
+@contextmanager
+def wait_for_rank_zero():
```

Comment:
(Discussed offline with hanlin, but would prefer that we make this shared helper more specific for torchvision, since this use case is very specific)

It's a much more common use case for rank zero to run some (extra) code, wait for a barrier, then all ranks run the same code rather than for rank zero to run code, wait, and then let all other ranks run the same code (while rank zero waits -- there should probably be another barrier at the end)